internal/dxcore/dxcore.c

```diff
@@ -239,7 +251,36 @@ static void dxcore_add_adapter(struct dxcore_context* pCtx, struct dxcore_lib* p
         log_infof("Adding new adapter via dxcore hAdapter:%x luid:%llx wddm version:%d", pAdapterInfo->hAdapter, *((unsigned long long*)&pAdapterInfo->AdapterLuid), wddmVersion);
 }
 
-static void dxcore_enum_adapters(struct dxcore_context* pCtx, struct dxcore_lib* pLib)
+static int dxcore_enum_adapters3(struct dxcore_context* pCtx, struct dxcore_lib* pLib)
+{
+        struct dxcore_enumAdapters3 params = {0};
+        unsigned int adapterIndex = 0;
+
+        params.Filter = ENUMADAPTER3_FILTER_COMPUTE_ONLY;
```

Comment:
As a matter of interest, how does this interact with adapters that would have been found using the previous implementation?

Reply:
> under which conditions should this function not be used?

`EnumAdapters3` was introduced in Windows 10, version 2004. To not break earlier versions, if this functions doesn't exist, we need to fallback to `EnumAdapters2`.

> how does this interact with adapters that would have been found using the previous implementation

This returns all the adapters returned previously and, additionally, compute-only adapters (it will skip display-only adapters, but those are not interesting for CUDA and would get filtered out later anyway).

Reply:
Ok. It wasn't clear that this would still return the OTHER adapters. Could we at least add a comment or something to indicate that this is the behaviour?

Reply:
Added comments to explain

Reply:
Thanks!